GameSync.Application/EmailInfrastructure/Services/EmailService.cs

```diff
@@ -0,0 +1,82 @@
+using EnsureThat;
+using FluentValidation;
+using FluentValidation.Results;
+using GameSync.Domain.Shared.Commands;
+using MailKit.Net.Smtp;
+using Microsoft.Extensions.Configuration;
+using MimeKit;
+
+namespace GameSync.Application.EmailInfrastructure;
+
+/// <summary>
+/// Provides functionality for sending emails.
+/// </summary>
+public class EmailService : IEmailService
+{
+    private readonly IConfiguration _configuration;
+    private readonly IValidator<SendEmailCommand> _validator;
+    private readonly string _emailPassword;
+    private readonly string _authLogin;
+    private readonly string _senderEmail;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EmailService"/> class.
+    /// </summary>
+    /// <param name="configuration">The application configuration.</param>
+    /// <param name="validator">The validator for the email command.</param>
+    /// <param name="emailPassword">Password to login in email account.</param>
+    /// <param name="authLogin">Login to authenticate account.</param>
+    /// <param name="senderEmail">Email address of sender message.</param>
+    public EmailService(IConfiguration configuration, IValidator<SendEmailCommand> validator, string emailPassword, string authLogin, string senderEmail)
+    {
+        _configuration = configuration;
+        _validator = validator;
+        _emailPassword = emailPassword;
+        _authLogin = authLogin;
+        _senderEmail = senderEmail;
+    }
+
+    /// <summary>
+    /// Sends an email based on the provided command.
+    /// </summary>
+    /// <param name="command">The command containing the email details.</param>
+    /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+    /// <returns>
+    /// A <see cref="CommandResult"/> representing the result of the email sending operation.
+    /// </returns>
+    public async Task<CommandResult> SendEmailAsync(SendEmailCommand command, CancellationToken cancellationToken)
```

Comment:
I don't think this is a good practice to use `Command` classes in `Infrastructure` layer. We should rather keep them separate.

Reply:
Ooook.. I see now. You copied the code from service to handler.. It shouldn't be that way. You should inject the `IEmailService` implementation into `SendEmailHandler`.